pkg/kubelet/probe.go

```diff
@@ -39,13 +42,37 @@ var (
 	tcprober   = tcprobe.New()
 )
 
-func (kl *Kubelet) probeContainer(p *api.Probe, podFullName string, podUID types.UID, status api.PodStatus, container api.Container) (probe.Result, error) {
+const defaultProbeTimeout = 1 * time.Second
+
+func (kl *Kubelet) probeContainer(p *api.Probe,
+	podFullName string,
+	podUID types.UID,
+	status api.PodStatus,
+	container api.Container,
+	dockerContainer *docker.APIContainers,
+	defaultResult probe.Result) (result probe.Result, err error) {
+	if p == nil {
+		return probe.Success, nil
```

Comment:
Why does this return Success rather than the defaultResult? In what scenario would defaultResult be failure?